packages/runtime/container-runtime/src/gc/gcTelemetry.ts

```diff
@@ -119,76 +124,111 @@ export class GCTelemetryTracker {
 			return false;
 		}
 
+		// Non-tombstone events are logged once per event per node. A unique id is generated by joining
+		// node state (inactive / sweep ready), node's id and usage (loaded / changed / revived).
 		if (this.loggedUnreferencedEvents.has(uniqueEventId)) {
 			return false;
 		}
 		return true;
 	}
 
 	/**
-	 * Called when a node is used. If the node is not active, log an event indicating object is used when its not active.
+	 * Called when a node is used. If the node is not active or tombstone, log telemetry indicating object is used
+	 * when it should not have.
 	 */
 	public nodeUsed(nodeUsageProps: INodeUsageProps) {
 		// If there is no reference timestamp to work with, no ops have been processed after creation. If so, skip
 		// logging as nothing interesting would have happened worth logging.
-		// If the node is not unreferenced, skip logging.
-		const nodeStateTracker = this.getNodeStateTracker(nodeUsageProps.id);
-		if (!nodeStateTracker || nodeUsageProps.currentReferenceTimestampMs === undefined) {
+		if (nodeUsageProps.currentReferenceTimestampMs === undefined) {
 			return;
 		}
-
-		// We log these events once per event per node. A unique id is generated by joining node state (inactive / sweep ready),
-		// node's id and usage (loaded / changed / revived).
-		const uniqueEventId = `${nodeStateTracker.state}-${nodeUsageProps.id}-${nodeUsageProps.usageType}`;
-		const nodeType = this.getNodeType(nodeUsageProps.id);
-		if (
-			!this.shouldLogNonActiveEvent(
-				nodeUsageProps.id,
-				nodeType,
-				nodeUsageProps.usageType,
-				nodeStateTracker,
-				uniqueEventId,
-			)
-		) {
+		// If the node's unreferenced state is not tracked or it's not tombstoned, there is nothing to log.
```

Comment:
This comment says "or" but the code says `&&`... I'm confused as to what the intent here is.

Reply:
Ok, I see - We may detect Tombstone usage before tracking is set up.  You called this out in the PR description.  I think this comment should be more explicit about the why.